doc/decisions/0_drafts/metakey_semantics.md

```diff
@@ -0,0 +1,191 @@
+# Metakey semantics
+
+## Problem
+
+Metakeys (i.e., keys with namespace `KEY_NS_META`) have special requirements beyond what other namespaces want.
+This is because metakeys are used for specifications.
+
+These extra requirements are:
+
+1. Copying metadata from `spec:/` keys to other namespaces must share as much memory as possible.
+   Without sharing memory there would be massive unnecessary duplication of data.
+2. It must be possible to identify whether a metakey was copied from `spec:/` key, or was set directly.
+   Additionally, it must be possible to detect if the value of a copied metakey has changed.
+3. Changing the value of a copied metakey must not affect the original `spec:/` key (and its metadata).
+
+Unrelated to the specification use case, metakeys must also be prevented from having metadata of their own.
+
+## Constraints
+
```

Comment:
> The `spec` plugin should be a normal plugin, not needing internal APIs.

Agree that it should not need private APIs, but it **is not** a normal plugin. It's a very specialized hook plugin. So I think providing special APIs just for `spec` (or an alternative implementation of the same hook) would be an option.

Reply:
```suggestion
- The API should be hard to misuse.
- Elektra should protect against incorrect operations, that would lead to undefined behavior.
- As a specialized hook plugin `spec` could make use of specialized APIs.
  However, the need for such internal APIs should be limited as much as possible.
```